lib/charms/glauth_k8s/v0/ldap.py

```diff
@@ -244,9 +245,10 @@ def serialize_bool(self, starttls: bool) -> str:
 
 
 class LdapProviderData(LdapProviderBaseData):
-    bind_dn: str
-    bind_password_secret: str
-    auth_method: Literal["simple"]
+    bind_dn: str = Field(frozen=True)
+    bind_password: str = Field(exclude=True)
+    bind_password_secret: Optional[str] = None
```

Comment:
Just a **NONBLOCKING** question: why `bind_password_secret` is changed to `Optional[str]`?